crates/revm/src/optimism/mod.rs

```diff
@@ -151,6 +167,44 @@ impl RethL1BlockInfo for L1BlockInfo {
     }
 }
 
+/// The Canyon hardfork issues an irregular state transition that force-deploys the create2
+/// deployer contract. This is done by directly setting the code of the create2 deployer account
+/// prior to executing any transactions on the timestamp activation of the fork.
+pub fn ensure_create2_deployer<DB>(
+    chain_spec: Arc<ChainSpec>,
+    timestamp: u64,
+    db: &mut revm::State<DB>,
+) -> Result<(), RethError>
+where
+    DB: revm::Database,
+{
+    if chain_spec.fork_timestamps.canyon.unwrap_or_default() == timestamp &&
```

Comment:
can't find the spec to reference, but i believe that fork activation of a timestamp based happens upon crossing the specific timestamps. so, in order to properly determine whether the timestamp-based fork  has been activated at a given block, you need to check that the fork is active at a given timestamp, but is not active at the timestamp of the parent block